includes/customer-functions.php

```diff
@@ -653,73 +653,21 @@ function edd_maybe_add_customer_address( $customer_id = 0, $data = array() ) {
 	// Check if this address is already assigned to the customer.
 	$address_exists = $customer_addresses->query( $address_to_check );
 	if ( ! empty( $address_exists ) ) {
+		if ( 'billing' === $address_to_check['type'] ) {
+			$customer_addresses->update_item( $address_exists[0], array( 'is_primary' => true ) );
```

Comment:
Similar as before, I agree that we should stick to using the helper functions like `edd_update_customer_address()` and `edd_add_customer_address()`.